### PR TITLE
Isolate forkchoice to blockchain package

### PR DESCRIPTION
**Pull request by potuz**
This PR isolates forkchoice to the blockchain package, this way consumers do not need to worry about locks. 

In particular, it removes the function `ForkChoicer()`  that exposed explicitly forkchoice to external packages. 

The blockchain package exposes an interface `blockchain.ForkchoiceFetcher` that wraps all externally used forkchoice functions in a self-locked manner. External callers do not have access to forkchoice directly, but rather have to call blockchain package to lock forkchoice..

### CODE REVIEW

File: beacon-chain/blockchain/chain_info.go [hunk 1]
```diff
@@ -484,3 +482,66 @@ func (s *Service) Ancestor(ctx context.Context, root []byte, slot primitives.Slo
 func (s *Service) SetGenesisTime(t time.Time) {
 	s.genesisTime = t
 }
+
```
Comment (nisdas) [inline beacon-chain/blockchain/chain_info.go:485]: I think all these methods specific to forkchoice should be in their own file ex: `forkchoice_chain_info` or something of that sort

Comment (potuz) [reply beacon-chain/blockchain/chain_info.go:485]: 👍

**Review by terencechain — APPROVED** [review]

**Review by terencechain — DISMISSED** [review]

File: beacon-chain/blockchain/chain_info_forkchoice.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"context"
+	"time"
+
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	ethpbv1 "github.com/prysmaticlabs/prysm/v4/proto/eth/v1"
+)
+
+// SetGenesisTime sets the genesis time of beacon chain.
+func (s *Service) SetGenesisTime(t time.Time) {
```
Comment (nisdas) [inline beacon-chain/blockchain/chain_info_forkchoice.go:13]: Is this really a forkchoice method ? Seems unrelated

**Review by nisdas — APPROVED** [review]
